ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ClassLoadInvoker.java

```diff
@@ -539,9 +540,13 @@ private ClassLoadContext createVariablesExecutionContext(
     @Override
     public List<String> availableImports() {
         // Imports with prefixes
+        List<String> initialImports = getInitialImports();
         List<String> importStrings = new ArrayList<>();
         for (QuotedIdentifier prefix : importsManager.prefixes()) {
-            importStrings.add(String.format("(%s) %s", prefix, importsManager.getImport(prefix)));
+            String importStatement = String.format("(%s) %s", prefix, importsManager.getImport(prefix));
+            if (!initialImports.contains(importStatement)) {
+                importStrings.add(importStatement);
+            }
```

Comment:
Can't this be handled in the `ImportsManager` itself? i.e., have separate APIs in `ImportsManager` which will return the imports with and without this import?

@kdsuneraavinash thoughts?

Reply:
If  I understand correctly, this implementation would hide `jballerina.java import` even if it is explicitly imported, no?

So I think the better way would be to modify `storeImport` in imports manager. 

https://github.com/ballerina-platform/ballerina-lang/blob/c5efa989cd35b9631916e6cf5de1fc399c94ba2c/ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/invoker/classload/ImportsManager.java#L160-L164

However, simply hiding this import can cause several other issues,
- users cant use `java` import prefix because there is this implicit import with the same prefix. So they will be shown an error if they try to add `import ballerina/jballerina/java` explicitly.
- users still be able to use `java` prefix in code.

So these cases also need to be handled. But I think it is possible to handle these cases just by modifying imports manager.